### PR TITLE
add connect insight event

### PR DESCRIPTION
**Pull request by makarandp0**
was [previously](https://github.com/twilio/twilio-video.js/pull/1268) 

send room/connect event to insights following schema specified here:
https://code.hq.twilio.com/client/rtc-insights-spec/blob/connect_again/schema/video/event/room/connect.json

with this the logged room/connect event looks like:
https://kibana.dev-us1.eak.twilio.com/video/goto/5c1027ea43adbecc45f03fb94b350ab3

burndown :
- [x] merge https://code.hq.twilio.com/client/rtc-insights-spec/pull/69
- [x] release rtc-insights-spec 1.0.3 
- [ ] re-deploy sdk-ws-gateway 
- [ ] verify
- [ ] merge this PR

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by makarandp0** [comment]
@manjeshbhargav - I have incorporated your feedback from https://github.com/twilio/twilio-video.js/pull/1268 in this PR. Please take another look

File: lib/util/index.js [hunk 1]
```diff
@@ -553,6 +553,59 @@ function valueToJSON(value) {
...
+  ['preferredAudioCodecs', 'preferredVideoCodecs'].forEach(prop => {
+    if (typeof connectOptions[prop] !== 'undefined') {
+      payload[prop] = JSON.stringify(connectOptions[prop]);
+    }
+  });
+
+  if (connectOptions.bandwidthProfile) {
+    payload.bandwidthProfileOptions = {};
+    ['mode', 'maxTracks', 'trackSwitchOffMode', 'dominantSpeakerPriority'].forEach(prop => {
+      if (typeof connectOptions.bandwidthProfile[prop] === 'number' || typeof connectOptions.bandwidthProfile[prop] === 'string') {
```
Comment (manjeshbhargav) [inline lib/util/index.js:592]: @makarandp0 At this point we would have already validated the bandwidth profile options object. So you just need to check the presence of the property.
```suggestion
      if (prop in connectOptions.bandwidthProfile) {
```

File: lib/util/index.js [hunk 2]
```diff
@@ -553,6 +553,59 @@ function valueToJSON(value) {
...
+  [['maxVideoBitrate'], ['maxAudioBitrate'], ['iceTransportPolicy'], ['region'], ['name', 'roomName']].forEach(([prop, eventProp]) => {
+    eventProp = eventProp || prop;
+    if (typeof connectOptions[prop] === 'number' || typeof connectOptions[prop] === 'string') {
+      payload[eventProp] = connectOptions[prop];
+    }
+  });
+
+  // array props stringified.
+  ['preferredAudioCodecs', 'preferredVideoCodecs'].forEach(prop => {
+    if (typeof connectOptions[prop] !== 'undefined') {
```
Comment (manjeshbhargav) [inline lib/util/index.js:584]:
```suggestion
    if (prop in connectOptions) {
```

File: lib/util/index.js [hunk 3]
```diff
@@ -553,6 +553,59 @@ function valueToJSON(value) {
...
+  });
+
+  if (connectOptions.bandwidthProfile) {
+    payload.bandwidthProfileOptions = {};
+    ['mode', 'maxTracks', 'trackSwitchOffMode', 'dominantSpeakerPriority'].forEach(prop => {
+      if (typeof connectOptions.bandwidthProfile[prop] === 'number' || typeof connectOptions.bandwidthProfile[prop] === 'string') {
+        payload.bandwidthProfileOptions[prop] = connectOptions.bandwidthProfile[prop];
+      }
+    });
+    if (connectOptions.bandwidthProfile.renderDimensions) {
```
Comment (manjeshbhargav) [inline lib/util/index.js:596]:
```suggestion
    if ('renderDimensions' in connectOptions.bandwidthProfile) {
```

**Review by manjeshbhargav — CHANGES_REQUESTED** [review]
@makarandp0 Implementation looks good. However, since there is a new event on the EventObserver, please document it and add a CHANGELOG.md entry.

**Review by manjeshbhargav — APPROVED** [review]

**Pull request merged**